Server/Ebenezer/User.cpp

```diff
@@ -11796,7 +11902,18 @@ bool CUser::RunEvent(const EVENT_DATA* pEventData)
 			case EXEC_LOG_COUPON_ITEM:
 				LogCoupon(pExec->m_ExecInt[0], pExec->m_ExecInt[1]);
 				break;
-//
+
+			case EXEC_GIVE_PROMOTION_QUEST:
```

Comment:
As above, this is not implemented. If it's not implemented, it shouldn't be silently ignoring it -- it should continue to feed us our unhandled opcode error as before.

So you should either implement this opcode or remove the case so it isn't ignoring it.

(I'm also realising that we have some wonky duplicated code here that needs to be revisited. It definitely doesn't behave like this in the official 1.298 binaries)